chore(docs): add workspace watching feature doc (#3487)
lerna/lerna: website/docs/features/workspace-watching.md

---
id: workspace-watching
title: Workspace Watching
type: recipe
---

# Workspace Watching

:::note
Workspace Watching is available as of Lerna 6.4.0.
:::

Lerna can watch for file changes within packages and automatically execute commands from the root of the repository. This is useful if you need to rebuild packages or rerun tests as you update files during your development workflow.

This replaces the need to manually set up watching for each package individually.

## Examples

Watch all packages and echo the package name and the files that changed:

```sh
$ lerna watch -- echo \$LERNA_PACKAGE_NAME \$LERNA_FILE_CHANGES
```

Watch all packages and run the "build" script on a package when a file within it changes:

```sh
$ lerna watch -- lerna run build --scope=\$LERNA_PACKAGE_NAME
```

> Watch a single package and run the "build" script on it when a file within it changes:

Reviewer: ```suggestion
Watch all packages and run the "build" script on everything affected by the changes:

```sh
$ lerna watch -- lerna run build --since
```

Watch a single package and run the "build" script on it when a file within it changes:
```

Author: Added ✅
